src/SensioLabs/Melody/Composer/Composer.php

```diff
@@ -79,6 +79,14 @@ public function getVendorDir()
         return end($outputByLines);
     }
 
+    public function configureRepositories(array $repositories, $dir)
+    {
+        if (!empty($repositories)) {
+            $composerJsonPath = $dir . '/composer.json';
+            file_put_contents($composerJsonPath, json_encode(array('repositories' => $repositories)));
```

Comment:
reply to my self, yes, I really think we should generate the whole composer.json (repositories and packages) and then run a `composer update` instead of `composer require ...`

BTW, the generated composer.json could be used to calculate the WorkingDirectory uniqueness